cmd/main.go

```diff
@@ -1,30 +1,44 @@
 package main
 
 import (
+	"encoding/json"
+	"flag"
 	"fmt"
-	"io/ioutil"
 	"log"
-	"net/http"
 	"os"
+	"weather"
 )
 
+type Response struct {
+	OneWord string  `json:"oneword"`
+	Celcius float64 `json:"celcius"`
+}
+
 func main() {
 
+	var r Response
+
+	location := flag.String("location", "", "a city")
+	flag.Parse()
+
+	if len(*location) == 0 {
+		fmt.Printf("please enter a location\n")
+		os.Exit(2)
+	}
+	flag.Parse()
+
 	token := os.Getenv("WEATHERAPP_TOKEN")
 	if len(token) == 0 {
 		fmt.Printf("please set a weatherapp token\n")
 		os.Exit(2)
 	}
 
-	//call open weather map and get something
-	resp, err := http.Get(fmt.Sprintf("https://api.openweathermap.org/data/2.5/weather?q=london&appid=%s", token))
+	resp := weather.CallUrl(token, *location)
```

Comment:
What do you think about moving this logic inside the `weather` package? It seems reasonable for `main` to assemble the necessary parameters (token, location) and then pass them to some function in `weather` that just returns the weather as a string, doesn't it?

Reply:
✅